### ISSUE TITLE
Failed to deploy using OpenLiberty Operator 0.8.1

### ISSUE THREAD

**Issue by zhengchang907** [OpenLiberty/open-liberty-operator#340]
### What did you do?

1. I created an AKS cluster
2. I installed the OpenLiberty Operator 0.8.1 using kustomize
3. I tried to deploy [my sample app (without db)](https://github.com/zhengchang907/open-liberty-on-aks/tree/main/javaee-app-simple-cluster) but failed
4. I verified it still work with version 0.8.0

I was following [this guide](https://docs.microsoft.com/en-us/azure/aks/howto-deploy-java-liberty-app?tabs=without-sql#install-open-liberty-operator), just using 0.8.1 instead of 0.8.0

### What did you expect to see?

Able to deploy successfully 

### What did you see instead?

```console
$ kubectl get openlibertyapplication javaee-cafe-cluster
NAME                  IMAGE                                         EXPOSED   RECONCILED   AGE
javaee-cafe-cluster   zchang0408acr.azurecr.io/javaee-cafe:1.0.25             False        5s

$ kubectl describe openlibertyapplication javaee-cafe-cluster
<----trimmed---->
Status:
  Conditions:
    Last Transition Time:  2022-04-08T09:03:46Z
    Message:               Service account javaee-cafe-cluster isn't ready
    Status:                False
    Type:                  Reconciled
  Image Reference:         zchang0408acr.azurecr.io/javaee-cafe:1.0.25
Events:
  Type     Reason           Age                      From                   Message
  ----     ------           ----                     ----                   -------
  Warning  ProcessingError  <invalid> (x9 over 22s)  open-liberty-operator  Service account javaee-cafe-cluster isn't ready
```

### Environment

OpenLiberty Operator 0.8.1

**Comment by zhengchang907** [comment]
Tried to install using `kubectl` but also failed in app deployment step, same error message

**Comment by leochr** [comment]
@zhengchang907 To resolve some timing issues with pulling the image from the internal image registry, we added a check (included with 0.8.1) to ensure that the ServiceAccount includes an image pull secret. In OpenShift, the ServiceAccount automatically gets an image pull secret. In your environment, it seems that the ServiceAccount doesn't have a secret. Please confirm if that's the case. Is that expected? If so, wondering how the images are pulled from the internal registry, assuming there is one. Thanks.

**Comment by leochr** [comment]
@zhengchang907 Thank you. We'll need to fix the service account validation logic to account for this. It'll be released under 0.8.2. Let you know when it's available.

**Comment by zhengchang907** [comment]
@leochr Thank you, looking forward to it

**Comment by bconey** [comment]
Just wanted to add that this same behavior was seen when using the 0.8.1 operator on an EKS cluster using a private registry.  The secret for the registry was created in the namespace where the application was to be deployed, but the same errors as noted above for the service account were logged.  After adding the secret to the service account, the deployment completed successfully.

**Comment by dfateyev** [comment]
Please consider to release the updated version with the fix. The current issue blocks using the operator in all Kubernetes clusters / providers where are no pull secrets provided automatically.

**Comment by leochr** [comment]
We expect to have the release by the end of this month

**Comment by dfateyev** [comment]
@leochr is there any news on the release with the fix? Thanks!

**Comment by zhengchang907** [comment]
Hi @leochr,  any  updates?

**Comment by majguo** [comment]
Hello @leochr I'm teammate of @zhengchang907 , I noticed that [0.8.2](https://github.com/OpenLiberty/open-liberty-operator/tree/main/deploy/releases/0.8.2#installation) is released and also verified that the issue @zhengchang907 reported has been fixed in 0.8.2. Thanks for the fixing.

**Comment by leochr** [comment]
I was about to update this issue regarding the 0.8.2 release. Thanks for the confirmation @majguo
